src/fragmentizer.rs

```diff
@@ -0,0 +1,306 @@
+#[derive(Debug, Clone, Copy, PartialEq)]
+#[pyclass(name = "LineFragmentInfo", eq)]
+pub(crate) struct PyLineFragmentInfo {
+    start: usize,
+    end: usize,
+    announcement: Option<PyLiteralAnnouncement>,
+    ending: PyLineEnding,
+}
+
+impl From<PyLineFragmentInfo> for FragmentInfo {
+    fn from(value: PyLineFragmentInfo) -> Self {
+        Self::Line {
+            start: value.start,
+            end: value.end,
+            announcement: value.announcement.map(PyLiteralAnnouncement::into),
+            ending: value.ending.into(),
+        }
+    }
+}
+
+#[pymethods]
+impl PyLineFragmentInfo {
+    /// Create a new line fragment
+    #[new]
+    #[pyo3(signature = (start, end, announcement=None, ending=PyLineEnding::CrLf))]
+    pub(crate) fn new(
+        start: usize,
+        end: usize,
+        announcement: Option<PyLiteralAnnouncement>,
+        ending: PyLineEnding,
+    ) -> Self {
+        Self {
+            start,
+            end,
+            announcement,
+            ending,
+        }
+    }
+
+    /// Retrieve the start index of the detected line fragment
+    #[getter]
+    pub(crate) fn start(&self) -> usize {
+        self.start
+    }
+
+    /// Retrieve the end index of the detected line fragment
+    #[getter]
+    pub(crate) fn end(&self) -> usize {
+        self.end
+    }
+
+    /// Retrieve the literal announcement of the detected line fragment
+    #[getter]
+    pub(crate) fn announcement(&self) -> Option<PyLiteralAnnouncement> {
+        self.announcement
+    }
+
+    /// Retrieve the line ending of the detected line fragment
+    #[getter]
+    pub(crate) fn ending(&self) -> PyLineEnding {
+        self.ending
+    }
+}
+
+/// Python class describing a literal fragment detected by the fragmentizer
+#[derive(Debug, Clone, Copy, PartialEq)]
+#[pyclass(name = "LiteralFragmentInfo", eq)]
+pub(crate) struct PyLiteralFragmentInfo {
+    start: usize,
+    end: usize,
+}
+
+impl From<PyLiteralFragmentInfo> for FragmentInfo {
+    fn from(value: PyLiteralFragmentInfo) -> Self {
+        Self::Literal {
+            start: value.start,
+            end: value.end,
+        }
+    }
+}
+
+#[pymethods]
+impl PyLiteralFragmentInfo {
+    /// Create a new literal fragment
+    #[new]
+    pub(crate) fn new(start: usize, end: usize) -> Self {
+        Self { start, end }
+    }
+
+    /// Retrieve the start index of the detected literal fragment
+    #[getter]
+    pub(crate) fn start(&self) -> usize {
+        self.start
+    }
+
+    /// Retrieve the end index of the detected literal fragment
+    #[getter]
+    pub(crate) fn end(&self) -> usize {
+        self.end
+    }
+}
+
+/// Python class representing a fragmentizer
+#[derive(Debug, Clone)]
+#[pyclass(name = "Fragmentizer")]
+pub(crate) struct PyFragmentizer(Fragmentizer);
+
+#[pymethods]
+impl PyFragmentizer {
+    /// Create a new fragmentizer
+    #[new]
+    #[pyo3(signature = (*, max_message_size=None))]
```

Comment:
If I remove the `None`, is the python user forced to specify a value? I want to nudge the user to make a conscious decision here.